### ISSUE TITLE
Slow frame-rate

### ISSUE THREAD

**Issue by IanWitham** [kcunning/Katie-s-Rougish-PyGame#29]
Hi Katie. I was wondering why the frame-rate seemed to be so slow. This is very obvious when you see the lag between moving the mouse, and watching the "selection square" try to keep up. At first I thought it was because you redraw the whole level each time, but even then it shouldn't be so slow.

Anyway I think I have found the culprit. It is because every time you blit a monster, treasure, or wall tile you start by reloading the graphic from disk. You should only be loading the graphics once and then grabbing these graphics from memory when you need them.

So, for example, instead of this:

``` python
def draw_walls(self, walls, filename):
    ''' Draws walls on the game map
    '''
    for row in range(ROWS):
        for col in range(COLUMNS):
            if walls[row][col] != 0:
                wall = pygame.image.load(IMG_DIR + filename)   # NOT GOOD!
                self.screen.blit(wall, (row*TILE_SIZE, col*TILE_SIZE))
```

You should have something like this:

``` python
def __init__(self):
    self.my_reusable_wall_tile_graphic = pygame.image.load(IMG_DIR + "my_wall_tile.png")   # I'll just keep this for later

def draw_walls(self, walls):
    ''' Draws walls on the game map
    '''
    for row in range(ROWS):
        for col in range(COLUMNS):
            if walls[row][col] != 0:
                self.screen.blit(self.my_reusable_wall_tile_graphic, (row*TILE_SIZE, col*TILE_SIZE)) # MUCH FASTER!
```

**Comment by Technohazard** [comment]
Hi guys! Another thing that might be causing framerate slowdown is re-rendering the text for your fonts in realtime.
Try rendering font 'labels' or commonly used text to its own surface, then only re-rendering it before blit if the value changes.

Another thing that helps framerate is decoupling the game engine from the rendering engine (as Ian mentioned in another issue) 

I like the pygame.sprite.Sprite class http://www.pygame.org/docs/ref/sprite.html which allows you to automate a lot of stuff by simply extending your classes (monster, player, tile) from the base "Sprite" class, then adding them to 'rendering groups.'

Nice work so far, I enjoy following this project and the blog entries (saw this on Reddit). :)

**Comment by IanWitham** [comment]
@Technohazard Yes there is definitely more optimization that could be done around only updating that-which-has-changed on each frame. However PyGame should be able to easily handle the current graphics rendering load, which leads me to think that the continual loading of graphics from disk is the major problem at the moment.

Also a roguelike shouldn't need a _super_ high frame rate. I only raise it as an issue at this early stage because the frame rate is _so_ low -- like 2 or 3 frames per second perhaps.

On a side note, your game projects look really cool. I look forward to having a better look at them .

**Comment by IanWitham** [comment]
This has been fixed in my new pull request (issue #30)
